### PR TITLE
SDK documentation improvement

### PR DESCRIPTION
**Pull request by Jasminna**
## OVERVIEW

- Typedoc version upgraded to 0.22.4
- The view of our generated docs is changed, due to breaking changes introduces in typedoc 0.20.0 version
- Moved the directDebitSubscribe and idealSubscribe methods in subscription module and added deprecated description next to them in payments module
- Added more comprehensive description for methods parameters and responses 
- Removed redundant getPaymentTools method

## WHERE SHOULD THE REVIEWER START?

_`.github/workflows/build.yml`_
_`clientside_test/runner.js`_
_`index.d.ts`_
_`package.json`_
_`typedoc.json`_
_`/src`_

## ANY NEW DEPENDENCIES ADDED?

_List any new dependencies added._

- [ ] Does it work in IE >= 11?
- [ ] _Does it work in other browsers?_

## CHECKLIST

_Be sure all items are_ ✅ _before submitting a PR for review._

- [ ] Verify you bumped the lib version according to [Semantic Versioning Standards](http://semver.org)
- [ ] Verify you updated the CHANGELOG
- [ ] Verify this branch is rebased with the latest master

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -1,107 +1,109 @@
...
+  signUpV2: 'v2/accounts',
   signIn: '/accounts/authenticate',
   signInV2: '/v2/accounts/authenticate',
   signOut: '/accounts/logout',
-  signUp: '/accounts',
-  signUpV2: 'v2/accounts',
+  changePassword: '/accounts/change-password',
   requestNewPassword: '/accounts/forgot-password',
-  setNewPassword: (token: any) => `/accounts/forgot-password/${token}`,
+  setNewPassword: (token: string): string => `/accounts/forgot-password/${token}`,
```
Comment (mirovladimitrovski) [inline src/constants/index.ts:10]: You don't need the `string` return type, it's implicit by the string result that this function returns.

File: src/endpoints/branding.ts [hunk 2]
```diff
@@ -14,18 +16,42 @@ class Branding extends BaseExtend {
...
+   *    paywall_footer: string;
+   *    default: boolean;
+   *    template_id: number;
+   *    terms_url: string;
+   *    dark_mode: boolean;
+   * }
+   * ```
    */
-  async getBranding(clientId: string, brandingId: string | number) {
+  async getBranding(clientId: string, brandingId = 'default'): Promise<AxiosResponse<BrandingDetails>> {
```
Comment (mirovladimitrovski) [inline src/endpoints/branding.ts]: This implies that `brandingId` will always be a string, although apart from the `default` value, it's a number. Here's one way we can express that:

```suggestion
  async getBranding(clientId: string, brandingId: number | 'default' = 'default'): Promise<AxiosResponse<BrandingDetails>> {
```

**Review by hf — DISMISSED** [review]
Looks good.

**Review by mirovladimitrovski — DISMISSED** [review]

File: src/constants/index.ts (hunk 1 above)
Comment (Jasminna) [reply src/constants/index.ts:10]: It shows warning if return type is not provided.

**Review by hf — APPROVED** [review]

**Pull request merged**